Src/main.c

```diff
@@ -542,12 +545,23 @@ void StartDefaultTask(void const* argument)
 
     /* USER CODE BEGIN 5 */
     /* Infinite loop */
-    uint32_t prevWakeTime = osKernelSysTick();
+    HAL_GPIO_WritePin(LED2_GPIO_Port, LED2_Pin, 0);
 
     for (;;)
     {
-        osDelayUntil(&prevWakeTime, 250);
-        HAL_GPIO_TogglePin(LED2_GPIO_Port, LED2_Pin);
+        osDelay(500);
+
+        // blink once for PRELAUNCH phase
+        // blink twice for BURN phase
+        // blink 3 times for COAST phase
+        // blink 4 times for DROGUE_DESCENT phase
+        // blink 5 times for MAIN_DESCENT phase
+        for (int i = -1; i < currentFlightPhase; i++)
+        {
+            HAL_GPIO_WritePin(LED2_GPIO_Port, LED2_Pin, 1);
+            osDelay(100);
+            HAL_GPIO_WritePin(LED2_GPIO_Port, LED2_Pin, 0);
```

Comment:
Need to add delay after LED2 is turned off because otherwise the LED just stays on.